### ISSUE TITLE
Adding front matter to help promote the latest version of Zowe Docs articles in Google searches

### ISSUE THREAD

**Issue by anaxceron** [zowe/docs-site#3527]
## Is your request for enhancement related to a problem? Please describe.
Currently, Google is indexing older versions of Zowe Docs articles, which means that the latest version of articles are not included in search results. This can lead to Zower users getting outdated information if they're not aware that they've landed on an older version of Zowe Docs.

## Describe the solution you'd like
Apply annotations to help alert Google that older versions of Zowe Docs articles should be ignored when a latest version is available.

## Related doc pages
https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls

## Additional context
<img width="734" alt="image" src="https://github.com/zowe/docs-site/assets/109533923/f025df82-b21e-4f24-9f08-b656b6138106">

**Comment by t1m0thyj** [comment]
Since removing old version links from our sitemap in https://github.com/zowe/docs-site/pull/3329 didn't help, we'll likely need to add metadata to each package as mentioned above. Rather than manually adding a "canonical" link as front matter for each page which would be tedious, we should be able to model after https://github.com/camunda/camunda-docs/pull/2577 which solved the problem by swizzling the Docusaurus `Metadata` component so that a "canonical" link is automatically generated for each old version page.

**Comment by Maanvi212006** [comment]
Hi @anaxceron, I've been looking into this issue and have a good understanding of the approach (swizzling the Docusaurus Metadata component, similar to what camunda/camunda-docs#2577 did). I'd love to take a stab at implementing this. Is this still being actively worked on, or would you be open to assigning it to me?

**Comment by anaxceron** [comment]
> Hi [@anaxceron](https://github.com/anaxceron), I've been looking into this issue and have a good understanding of the approach (swizzling the Docusaurus Metadata component, similar to what [camunda/camunda-docs#2577](https://github.com/camunda/camunda-docs/pull/2577) did). I'd love to take a stab at implementing this. Is this still being actively worked on, or would you be open to assigning it to me?

Hey @Maanvi212006. If you'd like to take a crack at this, please go right ahead. And no, no one is currently working on this. Lmk if you have any other questions.

**Comment by Maanvi212006** [comment]
Okay @anaxceron 
I will go through it and then open a PR soon

**Comment by Maanvi212006** [comment]
Hi @anaxceron ,
I'm following the approach from [camunda/camunda-docs#2577](https://github.com/camunda/camunda-docs/pull/2577).

I have used the same structure as used in the camunda PR : 
src/theme/DocItem/Metadata folder with 3 files:
1. index.js
2. determineCanonical.js
3. determinCanonical.spec.js

However, even a simple console.log() at the top of the custom MetadataWrapper is not appearing in the browser console, meaning the swizzled component is not being used.

Tried:
- Disabling experimental_faster
- `yarn swizzle @docusaurus/theme-classic DocItem/Metadata --wrap` to let Docusaurus create the proper wrapper

Any idea why the swizzled DocItem/Metadata might not be picked up? Or is there something special about how docs are structured in this repo?
Happy to share my files if needed.
Thanks!

**Comment by anaxceron** [comment]
> Hi [@anaxceron](https://github.com/anaxceron) , I'm following the approach from [camunda/camunda-docs#2577](https://github.com/camunda/camunda-docs/pull/2577).
> 
> I have used the same structure as used in the camunda PR : src/theme/DocItem/Metadata folder with 3 files:
> 
> 1. index.js
> 2. determineCanonical.js
> 3. determinCanonical.spec.js
> 
> However, even a simple console.log() at the top of the custom MetadataWrapper is not appearing in the browser console, meaning the swizzled component is not being used.
> 
> Tried:
> 
> * Disabling experimental_faster
> * `yarn swizzle @docusaurus/theme-classic DocItem/Metadata --wrap` to let Docusaurus create the proper wrapper
> 
> Any idea why the swizzled DocItem/Metadata might not be picked up? Or is there something special about how docs are structured in this repo? Happy to share my files if needed. Thanks!

@Maanvi212006 Not sure what's going on. I can ask a dev to take a look, this is beyond my expertise. Do you have a branch you could share?

**Comment by Maanvi212006** [comment]
Hi @anaxceron 

I pushed my current work to my fork:
Branch : feature/canonical-urls 
Link : https://github.com/Maanvi212006/zowe-docs-site/tree/feature/canonical-urls

The main blocker is that the swizzled Metadata component doesn't seem to be running at all (even console.log at the top doesn't appear in browser console).

Path : C:\Users\Owner\zowe-docs-site\src\theme\DocItem\Metadata
Pls do let me know.
Thanks

**Comment by Maanvi212006** [comment]
Hi @anaxceron , just following up on this. Has a dev had a chance to look at it yet? Happy to provide any additional context or make changes based on feedback. 
Thanks!

**Comment by anaxceron** [comment]
> Hi [@anaxceron](https://github.com/anaxceron) , just following up on this. Has a dev had a chance to look at it yet? Happy to provide any additional context or make changes based on feedback. Thanks!

No, unfortunately we haven't had the time. We're working on some other higher priority items, but the goal is to look into this soon. You'll be notified when this happens.
